### ISSUE TITLE
[MNG-5503] Maven 3.1.0 fails to resolve artifacts produced by reactor build

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#6829]
**[Severin Gehwolf](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jerboaa)** opened **[MNG-5503](https://issues.apache.org/jira/browse/MNG-5503?redirect=false)** and commented

Maven 3.1.0 fails to resolve zip type dependencies produced via assemblies by modules higher up the reactor hierarchy. For example module A has a dep on module B of type zip. Module A produces a zip containing jars plus some other files which the build of module B should extract and copy to the output directory. However, dep solving of module B fails.

It works with Maven 3.0.5 but fails to build with Maven 3.1.0

---

**Affects:** 3.1.0

**Attachments:**
- [reproducer-zip-type-deps.zip](https://issues.apache.org/jira/secure/attachment/12713327/reproducer-zip-type-deps.zip) (_1.81 kB_)

**Issue Links:**
- [MNG-5591](https://issues.apache.org/jira/browse/MNG-5591) Installing workspace reader triggers MNG-5503

- [MNG-5214](https://issues.apache.org/jira/browse/MNG-5214) Dependency resolution substitutes g:a:v:jar for j:a:v:something-else when something-else isn't in the reactor

2 votes, 8 watchers

**Comment by jira-importer** [comment]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** commented

This appears to me to be a blocking regression.

Your test case has some minor issues that make it harder to identify the root cause (would have been better if all the versions were -SNAPSHOT versions... but none-the-less it does identity a major regression IMHO.

Here is the build output:

```
$ mvn clean verify
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] module-a
[INFO] module-b
[INFO] parent
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building module-a 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ module-a ---
[INFO] Deleting /Users/stephenc/tmp/mng5503/rep/module-a/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ module-a ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/stephenc/tmp/mng5503/rep/module-a/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ module-a ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ module-a ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/stephenc/tmp/mng5503/rep/module-a/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:testCompile (default-testCompile) @ module-a ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ module-a ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ module-a ---
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] Building jar: /Users/stephenc/tmp/mng5503/rep/module-a/target/module-a-1.0.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-assembly-plugin:2.2-beta-5:single (assemble) @ module-a ---
[INFO] Reading assembly descriptor: zip.xml
[INFO] Building zip: /Users/stephenc/tmp/mng5503/rep/module-a/target/module-a-1.0.0-SNAPSHOT.zip
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building module-b 1.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] module-a .......................................... SUCCESS [1.852s]
[INFO] module-b .......................................... FAILURE [0.018s]
[INFO] parent ............................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.034s
[INFO] Finished at: Fri Aug 16 15:08:26 IST 2013
[INFO] Final Memory: 11M/208M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project module-b: Could not resolve dependencies for project org.fedoraproject.rhbz991454:module-b:jar:1.0.0-SNAPSHOT: Could not find artifact org.fedoraproject.rhbz991454:module-a:zip:1.0.0-SNAPSHOT -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :module-b
```

I have removed the debug output, but essentially what seems to be happening is that it is looking for the metadata for the artifact outside of the reactor in order to determine whether or not the artifact with that specific classifier is present. I suspect, but I have not proved, that if the metadata was present (e.g. from having run as far as the install phase) then the artifact would be resolved from the reactor... but I'd like confirmation of that before downgrading this issue from blocker.

Jason, assigning to you as you're running the 3.1.1 release

**Comment by jira-importer** [comment]
**[Mikolaj Izdebski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mizdebsk)** commented

This was fixed in commit 11f46bd.

**Comment by jira-importer** [comment]
**[Mikolaj Izdebski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mizdebsk)** commented

Yes, I believe the IT is the one attached here or its original variant attached in Red Hat Bugzilla.
